drivers/spi/spi_context.h

```diff
@@ -112,13 +112,29 @@ static inline void spi_context_release(struct spi_context *ctx, int status)
 static inline int spi_context_wait_for_completion(struct spi_context *ctx)
 {
 	int status = 0;
+	uint64_t ms;
+
+	ms = MAX(ctx->tx_len, ctx->rx_len) * 8 * 1000 / ctx->config->frequency;
+	ms += 200; /* tolerance */
```

Comment:
A few comments:

- I think this change should be justified better. For example, a case where it solves a problem
- Tolerance should not be hardcoded, probably a Kconfig value at least.
- `ms` variable name does not help understanding its purpose.
- `config->frequency` may not necessarily contain the actual SPI frequency (more reason to make tolerance not hardcoded)

Reply:
> * I think this change should be justified better. For example, a case where it solves a problem

You do not think #33192 is justification enough?

> * Tolerance should not be hardcoded, probably a Kconfig value at least.

I agree. Either Kconfig or spi-controller.yaml binding, so the driver can fill it out in `struct spi_context`.

> * `ms` variable name does not help understanding its purpose.
> * `config->frequency` may not necessarily contain the actual SPI frequency (more reason to make tolerance not hardcoded)

When would `config->frequency` not contain the actual SPI frequency?